crypto/rsa/rsa_ameth.c

```diff
@@ -152,14 +152,14 @@ static int rsa_priv_encode(PKCS8_PRIV_KEY_INFO *p8, const EVP_PKEY *pkey)
 
     if (rklen <= 0) {
         ERR_raise(ERR_LIB_RSA, ERR_R_MALLOC_FAILURE);
-        ASN1_STRING_free(str);
+        ASN1_STRING_clear_free(str);
         return 0;
     }
 
     if (!PKCS8_pkey_set0(p8, OBJ_nid2obj(pkey->ameth->pkey_id), 0,
                          strtype, str, rk, rklen)) {
         ERR_raise(ERR_LIB_RSA, ERR_R_MALLOC_FAILURE);
-        ASN1_STRING_free(str);
+        ASN1_STRING_clear_free(str);
```

Comment:
The rk will actually contain the sensitive data and it will be leaked here. Please add `OPENSSL_clear_free(rk, rklen);` here. Also the str here does not contain here anything sensitive - imo the PSS parameters of the private key are not really sensitive, so there is no point in using ASN1_STRING_clear_free on it.

Reply:
> The rk will actually contain the sensitive data and it will be leaked here. Please add `OPENSSL_clear_free(rk, rklen);` here. Also the str here does not contain here anything sensitive - imo the PSS parameters of the private key are not really sensitive, so there is no point in using ASN1_STRING_clear_free on it.

Updated,Thanks for the review.